tools/build/cbt/fs.js

```diff
@@ -7,87 +7,80 @@
 const fs = require('fs');
 const glob = require('./glob');
 
-const Flags = {
-  SOURCE: 1,
-  TARGET: 1 << 1,
-  FILE: 1 << 2,
-  GLOB: 1 << 3,
-};
-
 class File {
   constructor(path) {
     this.path = path;
-    this.stat = null;
   }
 
-  performStat() {
-    this.stat = fs.statSync(this.path);
+  get stat() {
+    if (this._stat === undefined) {
+      this._stat = stat(this.path);
+    }
+    return this._stat;
+  }
+
+  exists() {
+    return this.stat !== null;
   }
 
   get mtime() {
-    if (!this.stat) {
-      this.performStat();
+    return this.stat && this.stat.mtime;
+  }
+
+  touch() {
+    const time = new Date();
```

Comment:
One thing that might get us slightly better results is touching the file to the newest mtime of all its inputs, rather than the actual build time, and also only doing this if the mtime is indeed less than that newest mtime, letting the natural mtime stand if the build task didn't skip modification of the file

Reply:
Yeah, but it's more complicated that way, and I kinda like the blunt solution.